### DISCUSSION TITLE
Does WikiJS create local files when rendering?

### DISCUSSION THREAD

**Discussion by 4e576rt8uh9ij9okp** [requarks/wiki#6169] [Help / Questions]
Hello,
so I'm wondering how performant and efficient WikiJS is.
The fact that it's not really needed to re-render a whole page on each request... and using HTML files for pages allows the use of CDN for them, while you would need multiple servers to serve pages faster.

So how does the rendering work? Can I make a use of CDN or do I need servers in each country?

**Comment by NGPixel** [comment]
The rendered version of each page is stored on disk the first time its fetched. However, only the page content itself is saved. The UI and other page elements are still served and rendered on client-side. So you can't serve these cached files directly, as it is not the full page.

You could however make use of a reverse proxy like Cloudflare which can cache full pages and assets for a specified duration.

**Reply by 4e576rt8uh9ij9okp** [reply]
I understand, but using reverse proxy doesn't help because it's still not a full page and its only the content.
I wish there was some way to enable static page rendering or something.

**Reply by NGPixel** [reply]
The reverse proxy can cache the full page, not only the content.
